dlt/load/load.py

```diff
@@ -268,15 +269,15 @@ def load_single_package(self, load_id: str, schema: Schema) -> None:
                 # only update tables that are present in the load package
                 applied_update = job_client.update_storage_schema(only_tables=all_tables | dlt_tables, expected_update=expected_update)
                 # update the staging dataset
-                merge_jobs = self.get_new_jobs_info(load_id, schema, "merge")
-                if merge_jobs:
+                staging_table_jobs = self.get_new_jobs_info(load_id, schema, job_client.get_stage_dispositions())
+                if staging_table_jobs:
                     logger.info(f"Client for {job_client.config.destination_name} will start initialize STAGING storage")
                     job_client.initialize_storage(staging=True)
```

Comment:
MAYBE we could remove the staging flag from `initialize_storage` and `update_storage_schema` and have `with_staging` context manager on SqlJobClientBase (the refactor that I proposed previously?). that would remove all traces of staging from JobClientBase (the top level ABC)

Reply:
I have created a subclass that supports the staging interface, it could also be a mixin actually, not sure.

Reply:
Also there is one method left called drop_tables, which has a staging parameter, maybe this should also be handled within the context manager?

Reply:
Ok, I removed it there too..